### PR TITLE
Refactor makefile targets

### PR DESCRIPTION
**Pull request by btoll**
## Summary

This PR removes the `OS_TYPE` and `ARCH` variables from the paths in `Makefile.mule`.

Note that this currently only supports linux/amd64.  The next step will be to run all tasks on all supported architectures.

For the most part, this is the same as the following PR, which I closed when I began this one:
https://github.com/algorand/go-algorand/pull/1070

## Test Plan

Pull down the branch and run the mule tasks.

### CODE REVIEW

File: scripts/release/mule/Makefile.mule [hunk 1]
```diff
@@ -46,30 +46,19 @@ ci-build: ci-setup buildsrc gen
 		 NO_BUILD=True VARIATIONS=$(OS_TYPE)/$(ARCH) \
 		 scripts/build_packages.sh $(OS_TYPE)/$(ARCH)
 
-package-deb:
-	scripts/release/mule/package/$(OS_TYPE)/$(ARCH)/deb/package.sh $(OS_TYPE) $(ARCH) $$WORKDIR
+# Builds targets from the sub-directories of ./scripts/release/mule/, such as `mule-package`, `mule-sign`, `mule-test`.
+# https://scene-si.org/2019/12/04/make-dynamic-makefile-targets/
+mule = $(shell ls -d scripts/release/mule/*/ | awk 'BEGIN { FS="/" ; OFS="-" } { print $$3, $$4 }')
```
Comment (bricerisingalgorand) [inline scripts/release/mule/Makefile.mule:51]: what does this do?

Comment (bricerisingalgorand) [reply scripts/release/mule/Makefile.mule:51]: By that I mean the awk portion.

Comment (btoll) [reply scripts/release/mule/Makefile.mule:51]: It parses the output of `ls` into the target into the target prefixes.
```
$ ls -d scripts/release/mule/*/
scripts/release/mule/package/  scripts/release/mule/sign/  scripts/release/mule/test/

$ ls -d scripts/release/mule/*/ | awk 'BEGIN { FS="/" ; OFS="-" } { print $3, $4 }'
mule-package
mule-sign
mule-test
```

File: scripts/release/mule/Makefile.mule [hunk 2]
```diff
@@ -46,30 +46,19 @@ ci-build: ci-setup buildsrc gen
...
-	scripts/release/mule/sign/$(OS_TYPE)/$(ARCH)/sign.sh $(OS_TYPE) $(ARCH) $(ARCH_BIT) $(VER) tar.gz $$WORKDIR
-
-package-sign-source:
-	scripts/release/mule/sign/$(OS_TYPE)/$(ARCH)/sign.sh $(OS_TYPE) $(ARCH) $(ARCH_BIT) $(VER) source $$WORKDIR
-
-package-test-deb:
-	scripts/release/mule/test/$(OS_TYPE)/$(ARCH)/test.sh $$WORKDIR $(OS_TYPE) $(ARCH) $(ARCH_BIT) $(VER) deb
-
-package-test-rpm:
-	scripts/release/mule/test/$(OS_TYPE)/$(ARCH)/test.sh $$WORKDIR $(OS_TYPE) $(ARCH) $(ARCH_BIT) $(VER) rpm
```
Comment (egieseke) [inline scripts/release/mule/Makefile.mule:74]: should the old package.sh, and sign.sh files be removed?

File: scripts/release/mule/Makefile.mule (hunk 1 above)
Comment (btoll) [reply scripts/release/mule/Makefile.mule:51]: The article this is based on that I linked to in the comments in the code uses `sed`, but I find `awk` to be easier to use.

File: scripts/release/mule/Makefile.mule (hunk 2 above)
Comment (btoll) [reply scripts/release/mule/Makefile.mule:74]: I'm not sure what you mean, nothing has been removed.  The files were moved out of the `linux/amd64` dirs.

Comment (btoll) [reply scripts/release/mule/Makefile.mule:74]: Reading this again, I may have misunderstood the question.  Does that answer it?

File: scripts/release/mule/Makefile.mule (hunk 1 above)
Comment (onetechnical) [reply scripts/release/mule/Makefile.mule:51]: Is this a make target? Shouldn't that be a `:`?

File: package-sign.yaml [hunk 3]
```diff
@@ -55,7 +55,7 @@ tasks:
         $XDG_RUNTIME_DIR/gnupg/S.gpg-agent:/root/.gnupg/S.gpg-agent,
         $HOME/.gnupg/pubring.kbx:/root/.gnupg/pubring.kbx
       ]
-    target: package-sign-tarball WORKDIR=/projects/go-algorand
+    target: mule-sign-tar.gz WORKDIR=/projects/go-algorand
```
Comment (onetechnical) [inline package-sign.yaml:58]: Technically this is fine, stylistically the `.` in `tar.gz` kind of hurts my eyes...I'd prefer `targz`, but this is non-blocking.

File: scripts/release/mule/Makefile.mule (hunk 1 above)
Comment (btoll) [reply scripts/release/mule/Makefile.mule:51]: No, that's not a target.

File: scripts/release/mule/Makefile.mule (hunk 2 above)
Comment (egieseke) [reply scripts/release/mule/Makefile.mule:74]: I see that they were moved, ok thanks.

File: package-sign.yaml (hunk 3 above)
Comment (btoll) [reply package-sign.yaml:58]: No, that's a variable.

**Pull request merged**